### ISSUE TITLE
[Bug] 更新之后出现grad_norm 和 loss nan错误

### ISSUE THREAD

**Issue by yy9996** [InternLM/InternLM#234]
### Describe the bug

更新之后出现grad_norm 和 loss nan错误
![QQ图片20230825135235](https://github.com/InternLM/InternLM/assets/102511364/f7bb9e32-a413-4b7c-b260-7091cde88e29)

### Environment

torch                         2.1.0.dev20230717+cu121
torch-scatter                 2.1.1
torchaudio                    2.1.0.dev20230717+cu121
torchmetrics                  0.11.4
torchvision                   0.16.0.dev20230717+cu121
tornado                       6.3.2
tqdm                          4.65.0
traitlets                     5.9.0
transaction                   3.1.0
transformers                  4.31.0

**Comment by yingtongxiong** [comment]
你好，请问是在哪个分支呢？以及具体的model config文件能分享一下吗？我在develop分支和main分支都没有复现出你这个结果。

**Comment by yy9996** [comment]
> 你好，请问是在哪个分支呢？以及具体的model config文件能分享一下吗？我在develop分支和main分支都没有复现出你这个结果。

是main分支，退回上个版本后就没这个问题了。下面是我的model config文件的内容：

JOB_NAME = "7b_train"

SEQ_LEN = 2048
HIDDEN_SIZE = 4096
NUM_ATTENTION_HEAD = 32
MLP_RATIO = 8 / 3
NUM_LAYER = 32
VOCAB_SIZE = 103168  

MODEL_ONLY_FOLDER = "local:/home/asus/llm/internlm-chat-7b/InternLM/initial_model_only_folder"
SAVE_CKPT_FOLDER = "local:/home/asus/llm/internlm-chat-7b/InternLM/trained_model/llm_ckpts_chat_lr1e-6_ordered_modified_test"

ckpt = dict(
    # Path to save training ckpt.
    save_ckpt_folder=SAVE_CKPT_FOLDER,
    # Path to continue training ckpt (load model weights and scheduler/context states).
    load_ckpt_folder=None,
    # Path to initialize with given model weights.
    load_model_only_folder=MODEL_ONLY_FOLDER,
    checkpoint_every=800,
    # Wheter to load optimizer states when continuing training.
    # load_optimizer=True,
)

TRAIN_FOLDER = "/home/asus/llm/data/data/internlm/ordered/removeextra/train"

data = dict(
    seq_len=SEQ_LEN,
    # micro_num means the number of micro_batch contained in one gradient update
    micro_num=1,
    # packed_length = micro_bsz * SEQ_LEN
    micro_bsz=4,
    pack_sample_into_one=False,
    total_steps=3600,
    skip_batches="",
    rampup_batch_size="",
    # Datasets with less than 50 rows will be discarded
    min_length=50,
    train_folder=TRAIN_FOLDER,
)

grad_scaler = dict(
    fp16=dict(
        # the initial loss scale, defaults to 2**16
        initial_scale=2**16,
        # the minimum loss scale, defaults to None
        min_scale=1,
        # the number of steps to increase loss scale when no overflow occurs
        growth_interval=1000,
    ),
    # the multiplication factor for increasing loss scale, defaults to 2
    growth_factor=2,
    # the multiplication factor for decreasing loss scale, defaults to 0.5
    backoff_factor=0.5,
    # the maximum loss scale, defaults to None
    max_scale=2**24,
    # the number of overflows before decreasing loss scale, defaults to 2
    hysteresis=2,
)

hybrid_zero_optimizer = dict(
    # Enable low_level_optimzer overlap_communication
    zero_overlap_communication=True,
    # bucket size for nccl communication params
    reduce_bucket_size=512 * 1024 * 1024,
    # grad clipping
    clip_grad_norm=1.0,
)

loss = dict(
    label_smoothing=0,
)

adam = dict(
    lr=1e-4,
    adam_beta1=0.9,
    adam_beta2=0.95,
    adam_beta2_c=0,
    adam_eps=1e-8,
    weight_decay=0.01,
)

lr_scheduler = dict(
    total_steps=data["total_steps"],
    init_steps=0,  # optimizer_warmup_step
    warmup_ratio=0.30,
    eta_min=1e-5,
    last_epoch=-1,
)

beta2_scheduler = dict(
    init_beta2=adam["adam_beta2"],
    c=adam["adam_beta2_c"],
    cur_iter=-1,
)

model = dict(
    checkpoint=False,
    num_attention_heads=NUM_ATTENTION_HEAD,
    embed_split_hidden=True,
    vocab_size=VOCAB_SIZE,
    embed_grad_scale=1,
    parallel_output=True,
    hidden_size=HIDDEN_SIZE,
    num_layers=NUM_LAYER,
    mlp_ratio=MLP_RATIO,
    apply_post_layer_norm=False,
    dtype="torch.bfloat16",
    norm_type="rmsnorm",
    layer_norm_epsilon=1e-5,
)
"""
zero1 parallel:
    1. if zero1 <= 0, The size of the zero process group is equal to the size of the dp process group,
        so parameters will be divided within the range of dp.
    2. if zero1 == 1, zero is not used, and all dp groups retain the full amount of model parameters.
    3. zero1 > 1 and zero1 <= dp world size, the world size of zero is a subset of dp world size.
        For smaller models, it is usually a better choice to split the parameters within nodes with a setting <= 8.
pipeline parallel: pipeline parallel size, only 1 is accepted currently.
tensor parallel: tensor parallel size, usually the number of GPUs per node, only 1 is accepted currently.
"""
parallel = dict(
    zero1=7
    # pipeline=1,
    # tensor=8,
)
enable_tb = True  # Whether to enable writing to TensorBoard, defaults to True
tensorboard_folder = None  # The new log folder for TensorBoard, defaults to None
resume_tb_folder = None  # The old log folder for resuming checkpoint, defaults to None

cudnn_deterministic = False
cudnn_benchmark = False

**Comment by yingtongxiong** [comment]
这个是在几张卡上跑呢，我看你吧zero1设置成了7。

**Comment by yy9996** [comment]
> 这个是在几张卡上跑呢，我看你吧zero1设置成了7。

七张卡

**Comment by yingtongxiong** [comment]
你试试把config中的hybrid_zero_optimizer字典中的zero_overlap_communication=True, 这个参数设置成False，看看还会有这个问题吗？

**Comment by yingtongxiong** [comment]
请问你load的checkpoint是在更新后的main分支上save的吗？还是说是从哪里获取的？

**Comment by yingtongxiong** [comment]
![image](https://github.com/InternLM/InternLM/assets/45058324/b8bfa5ee-44f4-4637-b994-1644acbe5fc1)
我在和你同样的并行配置下，重新跑了一遍，loss是可以正常下降的。或许你可以再check一下你的checkpoint或者数据集。
